swim/update_rollup_test.go

```diff
@@ -99,11 +100,23 @@ func (s *RollupTestSuite) TestFlushesAfterInterval() {
 	})
 
 	s.NotNil(s.r.FlushTimer(), "expected timer to be set")
+
+	// This is a race condition which never got triggered yet. If the scheduler
+	// decides to yield at this comment and proceed with flushing the buffer,
+	// assertion below will fail. Oops.
 	s.Len(s.r.Buffer(), 2, "expected two updates to be in buffer")
 
-	time.Sleep(2 * time.Millisecond)
+	// Flushing the buffer used to take <2ms on an otherwise idle machine. On
```

Comment:
I'm unsure if the test is actually meant to assert that the buffer gets flushed after the `flushInterval`. We should check what contract `updateRollup` is providing for flushing and if this matters.

If the contract is strict (i.e. flushing *MUST* occur after the `flushInterval`), then the test is genuinely failing. I think it would be a pretty bad contract to provide though and impossible to guarantee.

Let's have a quick chat with @jwolski and try to find out how precise the flushes need to be for an `updateRollup`.